daemon/volumes_unix.go

```diff
@@ -92,6 +92,11 @@ func parseBindMount(spec, volumeDriver string) (*mountPoint, error) {
 	}
 
 	if len(source) == 0 {
+		//validate the name of named volume
+		if strings.HasPrefix(name, "./") || strings.HasPrefix(name, "../") {
```

Comment:
This seems to block out the bad stuff: `if regexp.MatchString("(^.+[^0-9A-Za-z_]+$)|(/)", name)`